src/basilisp/core.lpy

```diff
@@ -6675,6 +6675,16 @@
 
 (prefer-method evolve basilisp.lang.interfaces/IRecord basilisp.lang.interfaces/IPersistentMap)
 
+(defn -mangle-py-field
+  "Implementation detail of :lpy:fn:`defrecord` to support Python's managled field
+  names."
+  [type-sym field-name]
+  (if (and (.startswith field-name "__")
+           (not (.endswith field-name "__")))
+    (let [munged-type-name (.lstrip (munge type-sym) "_")]
+      (str "_" munged-type-name field-name))
+    field-name))
```

Comment:
This is exactly the name mangling logic used by Python, so we just apply it whenever accessing record fields.

Reply:
This is a problem for versions of `attrs` <= 22.1.0. I suppose I can update the minimum version to use the new field `alias`.